util/reflect.go

```diff
@@ -545,6 +548,108 @@ func forEachFieldInternal(ni *NodeInfo, in, out interface{}, iterFunction FieldI
 	return errs
 }
 
+// ForEachDataField iterates the value supplied and iterates iterFunction for
+// each data tree node found in the function. No schema information is required
+// to perform the iteration.
+func ForEachDataField(value, in, out interface{}, iterFunction FieldIteratorFunc) Errors {
+	if IsValueNil(value) {
+		return nil
+	}
+
+	return forEachDataFieldInternal(&NodeInfo{FieldValue: reflect.ValueOf(value)}, in, out, iterFunction)
+}
+
+func forEachDataFieldInternal(ni *NodeInfo, in, out interface{}, iterFunction FieldIteratorFunc) (errs Errors) {
+	if IsValueNil(ni) {
+		return
```

Comment:
(optional) Naked returns tend to be confusing, unless this is part of the local style I would recommend:

return nil

Here and in a few other locations.

Reply:
Agreed -- have fixed this to returning `nil` or `errs` explicitly where required. I can't remember why we made this choice before, so I also fixed the `forEachFieldInternal` implementation as well. Thanks.